src/node_i18n.cc

```diff
@@ -542,6 +542,14 @@ bool InitializeICUDirectory(const std::string& path) {
   return status == U_ZERO_ERROR;
 }
 
+void SetDefaultTimeZone(const char* tzid) {
+  UChar lbl[255];
+  UErrorCode status = U_ZERO_ERROR;
+  u_charsToUChars(tzid, lbl, strlen(tzid) + 1);
+  ucal_setDefaultTimeZone(lbl, &status);
```

Comment:
- It’s not obvious (and afaict also not documented) that `ucal_setDefaultTimeZone` is threadsafe
- `u_charsToUChars` can overflow the stack, which gives a nice security vulnerability for free…?

```suggestion
  MaybeStackBuffer<UChar, 256> lbl(strlen(zid) + 1);
  UErrorCode status = U_ZERO_ERROR;
  u_charsToUChars(tzid, lbl.out(), strlen(tzid) + 1);
  // This is threadsafe:
  ucal_setDefaultTimeZone(lbl.out(), &status);
```

Reply:
Oh, and I have absolutely no clue what `lbl` could stand for. :)

Reply:
> * It’s not obvious (and afaict also not documented) that `ucal_setDefaultTimeZone` is threadsafe

it's mutexed and threadsafe, and generally documented (not this fcn specifically)

> * `u_charsToUChars` can overflow the stack, which gives a nice security vulnerability for free…?

i guess lbl is label?  +1 for using a maybestackbuffer.

Reply:
Ah right, I forgot that `u_charsToUChars()` is susceptible to overflows. Good catch! And yeah, `lbl` is `label` but it really ought to be `id` ;-)